noria/src/data.rs

```diff
@@ -440,107 +425,120 @@ impl<'a> Into<Cow<'a, str>> for &'a DataType {
                     String::from_utf8_lossy(&bts[..])
                 }
             }
-            _ => panic!("attempted to convert a {:?} to a string", self),
+            _ => panic!("attempted to convert a {:?} to a string", data),
         }
     }
 }
 
-impl Into<String> for &'_ DataType {
-    fn into(self) -> String {
-        let cow: Cow<'_, str> = self.into();
+impl From<&'_ DataType> for String {
```

Comment:
I think this `impl` should probably be removed. Anyone who wants this should `.clone().into()`.

Reply:
If we remove this `impl`, then the next one (`impl From<DataType> for String`) will have to itself convert to a `Cow`, because both of them relied on the actual conversion defined by `Cow` at line 411. That is, I can remove that `impl`, but I would change

```
impl From<DataType> for String {
    fn from(data: DataType) -> Self {
        let cow: Cow<'_, str> = (&data).into();
        cow.to_string()
    }
}
```

Would that be better?

Reply:
I think the `impl` for `String` should have its own impl rather than going through `Cow`, because when you have an owned `DataType` you can often entirely avoid the allocation + memcpy that the `Cow` version requires.